ui/components/src/ThresholdsEditor/ThresholdsEditor.tsx

```diff
@@ -0,0 +1,238 @@
+  const handleThresholdValueChange = (e: React.ChangeEvent<HTMLInputElement>, i: number) => {
+    setSteps(
+      produce(steps, (draft) => {
+        const step = draft?.[i];
+        if (step) {
+          step.value = Number(e.target.value);
+        }
+      })
+    );
+  };
+
+  const handleThresholdColorChange = (color: string, i: number) => {
+    if (thresholds !== undefined) {
+      onChange(
+        produce(thresholds, (draft) => {
+          if (draft.steps !== undefined) {
+            const step = draft.steps[i];
+            if (step) {
+              step.color = color;
+            }
+          }
+        })
+      );
+    }
+  };
+
+  const handleDefaultColorChange = (color: string) => {
+    if (thresholds !== undefined) {
+      onChange(
+        produce(thresholds, (draft) => {
+          draft.default_color = color;
+        })
+      );
+    }
+  };
+
+  // sort thresholds in ascending order every time an input blurs
+  const handleThresholdBlur = () => {
+    if (steps !== undefined) {
+      const sortedSteps = [...steps];
+      sortedSteps.sort((a, b) => a.value - b.value);
+      if (thresholds !== undefined) {
+        onChange(
+          produce(thresholds, (draft) => {
+            draft.steps = sortedSteps;
+          })
+        );
+      }
+    }
+  };
+
+  const deleteThreshold = (i: number): void => {
+    if (thresholds !== undefined) {
+      const updatedThresholds = produce(thresholds, (draft) => {
+        if (draft.steps) {
+          draft.steps.splice(i, 1);
+        }
+      });
+      onChange(updatedThresholds);
+    }
+  };
+
+  const addThresholdInput = (): void => {
+    focusRef.current = true;
+    if (thresholds === undefined) {
+      onChange({
+        steps: [{ value: DEFAULT_STEP }],
+      });
+    } else {
+      onChange(
+        produce(thresholds, (draft) => {
+          const steps = draft.steps;
+          if (steps?.length) {
+            const lastStep = steps[steps.length - 1];
+            const color = palette[steps.length] ?? getRandomColor(); // we will assign color from the palette first, then generate random color
+            steps.push({ color, value: (lastStep?.value ?? 0) + DEFAULT_STEP }); // set new threshold value to last step value + 10
+          } else if (steps) {
+            steps.push({ value: DEFAULT_STEP });
+          }
+        })
+      );
+    }
+  };
+
+  const handleModeChange = (event: React.MouseEvent, value: string): void => {
+    if (thresholds !== undefined) {
+      onChange(
+        produce(thresholds, (draft) => {
+          draft.mode = value === 'Percent' ? 'Percent' : undefined;
+        })
+      );
+    }
+  };
+
+  return (
+    <OptionsEditorGroup
+      title="Thresholds"
+      icon={
+        <IconButton size="small" aria-label="add threshold" onClick={addThresholdInput}>
+          <PlusIcon />
+        </IconButton>
+      }
+    >
+      <ToggleButtonGroup
+        exclusive
+        value={thresholds?.mode ?? 'absolute'}
```

Comment:
Should this be `Absolute` to match the value below? I _think_ that might be why the button doesn't show selected in the screenshots I included as a top-level comment.

Reply:
thanks for catching this!